pygmt/helpers/utils.py

```diff
@@ -315,6 +315,108 @@ def non_ascii_to_octal(argstr):
     return argstr.translate(str.maketrans(mapping))
 
 
+def build_arg_list(
+    kwdict: dict,
+    confdict: dict | None = None,
+    infile: str | pathlib.PurePath | list[str | pathlib.PurePath] | None = None,
+    outfile: str | pathlib.PurePath | None = None,
+) -> list[str]:
+    r"""
+    Convert keyword dictionaries and input/output files into a list of GMT arguments.
+
+    Make sure all values in ``kwdict`` have been previously converted to a string
+    representation using the ``kwargs_to_strings`` decorator. The only exceptions are
+    ``True``, ``False`` and ``None``.
+
+    Any lists or tuples left will be interpreted as multiple entries for the same
+    command line option. For example, the kwargs entry ``"B": ["xa", "yaf"]`` will be
```

Comment:
Call it 'command-line option' or 'parameter'? Xref https://github.com/GenericMappingTools/pygmt/issues/886.

```suggestion
    Any remaining lists or tuples will be interpreted as multiple entries for the same
    parameter. For example, the kwargs entry ``"B": ["xa", "yaf"]`` will be
```